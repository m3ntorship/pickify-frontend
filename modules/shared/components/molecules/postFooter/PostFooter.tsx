import React from 'react';
import type { FC, ReactElement } from 'react';
import type { IPostFooter } from './IPostFooter';
import Share from '../../icons/share.svg';
import styles from './PostFooter.module.css';

const PostFooter: FC<IPostFooter.IProps> = ({
  numberOfVotes,
  showResult = false,
  handleClick,
}): ReactElement => {
  return (
    <div className={styles.container}>
      {!showResult && <span> Vote to uncover the total number of voters </span>}
      {showResult && <span>{numberOfVotes} votes</span>}
      <button type="button" onClick={handleClick}>
        <Share />
      </button>
    </div>
  );
};

export default PostFooter;
