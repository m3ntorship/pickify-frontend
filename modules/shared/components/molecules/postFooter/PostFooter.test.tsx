import { render, screen } from '@testing-library/react';
import type { TargetElement } from '@testing-library/user-event';
import userEvent from '@testing-library/user-event';
import * as React from 'react';
import * as renderer from 'react-test-renderer';
import PostFooter from './PostFooter';

describe('Testing Footer with snapshot', () => {
  it('Testing Footer with required props', () => {
    const tree = renderer
      .create(
        <PostFooter
          numberOfVotes={120}
          showResult
          sharePostHandler={(): boolean => true}
          postId=""
        />,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('Testing Footer with required props', () => {
    const tree = renderer
      .create(
        <PostFooter
          numberOfVotes={120}
          showResult
          sharePostHandler={(): boolean => true}
          postId=""
        />,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should call sharePostHandler when clicking on the share icon button', () => {
    const mockedFn = jest.fn();
    render(
      <PostFooter
        numberOfVotes={120}
        showResult
        sharePostHandler={mockedFn}
        postId=""
      />,
    );
    const shareButton: TargetElement = screen.getByTestId('share-button-icon');
    userEvent.click(shareButton);
    expect(mockedFn).toHaveBeenCalledTimes(1);
  });
});
