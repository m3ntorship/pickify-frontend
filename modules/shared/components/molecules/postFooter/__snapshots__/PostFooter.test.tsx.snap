// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Footer with snapshot Testing Footer with required props 1`] = `
<div
  className="container"
>
  <span>
    120
     votes
  </span>
  <button
    type="button"
  >
    <svg
      data-file-name="SvgShare"
    />
  </button>
</div>
`;

exports[`Testing Footer with snapshot Testing Footer with required props 2`] = `
<div
  className="container"
>
  <span>
    120
     votes
  </span>
  <button
    type="button"
  >
    <svg
      data-file-name="SvgShare"
    />
  </button>
</div>
`;
